rancher/versions-url-op

```diff
@@ -1 +1 @@
-awk -F "\"" '/tag_name/ {print $4}' | sed 's/v//' | sed '/-beta/d' | sort -V -r
+awk -F "git/refs/tags/" ' {print $2}' | awk -F "\"" '{print $1}' | awk NF | sed 's/v//' | grep -i -Ev "cr|rc|beta|alpha|pre|preview" | sort -V -r
```

Comment:
This filtering should be centralized and happen here:
https://github.com/devonfw/ide-mirrors/blob/master/get-versions

IMHO also the sorting should only happen in the centralized code so `versions-url-op` only contains the minimum required logic.
I already left the same suggestion in my review of PR #34

Reply:
I already noticed this but there is a small part here that cannot be applied on all tools. It's the removing of `v` because the problem is that how some tools are implemented is different from others. In some tools the `v-` prefix in versions is kept in `<tool>_VERSION` and added to the download `urls` as part of `<tool>_VERSION`. In others it was removed from `<tool>_VERSION` and it is hard coded in the `urls`. So a single logic that removes `v` or keep it won't work unless we also unify the logic in how we get `<tool>_VERSION` for all tools. This can be solved easily once we get rid of the legacy code and start fetching `<tool>_VERSION` directly from `available-versions` and this a new issue anyway

Reply:
I was not talking about `awk -F "git/refs/tags/" ' {print $2}' | awk -F "\"" '{print $1}' | awk NF | sed 's/v//'` but about `| grep -i -Ev "cr|rc|beta|alpha|pre|preview" | sort -V -r`.
IMHO the latter should be removed from all `version-url-op` files and only be done in the central `get-versions` script.

BTW: `preview` is already contained in `pre` so it can be omitted.

Reply:
I left a comment that this is still to be done in https://github.com/devonfw/ide/issues/918
So now, I can merge this PR.
This improvement can be implemented in another PR.
If you do not plan to implement this, simply unassign yourself from the issue (AFAIK you are now part of a different team).